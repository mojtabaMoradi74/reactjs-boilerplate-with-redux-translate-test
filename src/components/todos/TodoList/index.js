import React, { Children } from 'react';
import TodoItem from '../TodoItem';
import PropTypes from 'prop-types';
const TodoList = ({ todos, ...prop }) => {


	return (
		<ul className='list-group'>
			{
				Children.toArray(todos?.map((todo) => (
					<TodoItem id={todo.id} title={todo.title} completed={todo.completed} {...prop} />
				)))
			}
		</ul>
	);
};

TodoList.propTypes = {
	todos: PropTypes.array,

}

TodoList.defaultProps = {
	todos: [],
}

export default TodoList;
