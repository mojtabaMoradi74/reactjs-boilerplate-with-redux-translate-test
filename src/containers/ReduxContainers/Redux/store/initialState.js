import { uniqId } from "utils/uniqId";

export let todos_data = [
    {
        id: uniqId(),
        title: 'todo 1',
        completed: true,
    },
    {
        id: uniqId(),
        title: 'todo 2',
        completed: false,
    },
    {
        id: uniqId(),
        title: 'todo 3',
        completed: false,
    },
    {
        id: uniqId(),
        title: 'todo 4',
        completed: false,
    },
    {
        id: uniqId(),
        title: 'todo 5',
        completed: false,
    },
];


const initialState = {
    isFetching: true,
    errorMessage: "",
    data: todos_data,
};
export default initialState


