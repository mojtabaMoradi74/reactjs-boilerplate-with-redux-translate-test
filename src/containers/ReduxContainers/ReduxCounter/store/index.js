import countdown from "./countdown/reducer";
import counter from "./counter/reducer";

const count = {
    countdown,
    counter
}

export default count