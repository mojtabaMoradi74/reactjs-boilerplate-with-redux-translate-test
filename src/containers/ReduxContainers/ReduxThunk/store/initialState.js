import { uniqId } from "utils/uniqId";




const initialState = {
    isFetching: true,
    errorMessage: "",
    data: [],
};
export default initialState


