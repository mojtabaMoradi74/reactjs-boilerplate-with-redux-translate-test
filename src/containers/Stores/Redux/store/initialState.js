import { uniqId } from "utils/uniqId";

let todos = [
    {
        id: uniqId(),
        title: 'todo 1',
        completed: true,
    },
    {
        id: uniqId(),
        title: 'todo 2',
        completed: false,
    },
    {
        id: uniqId(),
        title: 'todo 3',
        completed: false,
    },
    {
        id: uniqId(),
        title: 'todo 4',
        completed: false,
    },
    {
        id: uniqId(),
        title: 'todo 5',
        completed: false,
    },
];


const initialState = {
    isFetching: true,
    errorMessage: "",
    data: todos,
};
export default initialState


