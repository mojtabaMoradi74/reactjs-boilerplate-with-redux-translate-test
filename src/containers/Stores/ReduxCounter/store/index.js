import countdown from "./countdown";
import counter from "./counter";

const count = {
    countdown,
    counter
}

export default count