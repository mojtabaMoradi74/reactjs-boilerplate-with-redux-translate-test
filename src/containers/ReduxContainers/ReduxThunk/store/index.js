import { createSlice } from "@reduxjs/toolkit";
import initialState from './initialState';
import { addTodoAsync, getTodosAsync, toggleCompleteAsync, deleteTodoAsync } from "containers/ReduxContainers/ReduxThunk/thunk";


const reduxTodoSlice = createSlice({
    name: "thunkTodo",
    initialState,
    reducers: {},
    extraReducers: {
        [getTodosAsync.fulfilled]: (state, action) => {
            console.log({ state, action }, { ...state, data: action?.payload?.todos }, 'getTodosAsync.fulfilled');
            return { ...state, data: action?.payload?.todos }
        },
        [addTodoAsync.fulfilled]: (state, action) => {
            state.data.push(action.payload.todo);
        },
        [toggleCompleteAsync.fulfilled]: (state, action) => {
            const index = state.data.findIndex((todo) => todo.id === action.payload.todo.id);
            state.data[index] = action.payload.todo;
        },
        [deleteTodoAsync.fulfilled]: (state, action) => {
            return { ...state, data: state.data.filter((todo) => todo.id !== action.payload.id) }
        },
    },
});

export const { addTodo, toggleComplete, deleteTodo } = reduxTodoSlice.actions;

export default reduxTodoSlice.reducer;