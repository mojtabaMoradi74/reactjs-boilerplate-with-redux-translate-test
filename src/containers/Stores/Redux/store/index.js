import { createSlice } from "@reduxjs/toolkit";
import initialState from './initialState';
import { addTodoFunc, toggleCompleteFunc, deleteTodoFunc } from './helperFunc';


const reduxTodoSlice = createSlice({
    name: "reduxTodo",
    initialState,
    reducers: {
        addTodo: addTodoFunc,
        toggleComplete: toggleCompleteFunc,
        deleteTodo: deleteTodoFunc,
    },
});

export const { addTodo, toggleComplete, deleteTodo } = reduxTodoSlice.actions;

export default reduxTodoSlice.reducer;