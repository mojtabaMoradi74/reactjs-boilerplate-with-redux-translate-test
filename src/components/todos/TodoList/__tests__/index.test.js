import { mount } from 'enzyme';
import TodoList from 'components/todos/TodoList';
import { todos_data } from 'containers/Stores/Redux/store/initialState';
import { checkProps, ReactIntProvider } from 'utils/testing';
// import renderer from 'react-test-renderer';

describe('TodoList component', () => {
    const items = todos_data;

    let wrapper;

    beforeEach(() => {
        wrapper = mount(ReactIntProvider(<TodoList todos={items} />))
    });

    describe('Checking PropTypes', () => {

        it('Should not throw a warning', () => {
            const propsErr = checkProps(TodoList, items)
            expect(propsErr).toBeUndefined();
        });

    });

    describe('Checking element', () => {

        it('Should not throw a warning', () => {
            let listUl = wrapper.find('.list-group');
            expect(listUl.is('ul')).toBe(true);
            expect(listUl.type()).toBe('ul');
            expect(listUl.children()).toHaveLength(items.length);


            // const renderedComponent = renderer.create(
            //     ReactIntProvider(<TodoList todos={items} />)
            // ).toJSON();

            // expect(renderedComponent).toMatchSnapshot();
        });

    });

})