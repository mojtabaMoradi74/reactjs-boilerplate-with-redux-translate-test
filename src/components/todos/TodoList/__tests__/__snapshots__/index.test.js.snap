// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component Checking element Should not throw a warning 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item list-group-item-success"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="d-flex align-items-center"
      >
        <input
          className="mr-3"
          defaultChecked={true}
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="ml-1 mr-1"
        >
          todo 1
        </span>
      </span>
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
    </div>
  </li>
  <li
    className="list-group-item false"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="d-flex align-items-center"
      >
        <input
          className="mr-3"
          defaultChecked={false}
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="ml-1 mr-1"
        >
          todo 2
        </span>
      </span>
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
    </div>
  </li>
  <li
    className="list-group-item false"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="d-flex align-items-center"
      >
        <input
          className="mr-3"
          defaultChecked={false}
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="ml-1 mr-1"
        >
          todo 3
        </span>
      </span>
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
    </div>
  </li>
  <li
    className="list-group-item false"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="d-flex align-items-center"
      >
        <input
          className="mr-3"
          defaultChecked={false}
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="ml-1 mr-1"
        >
          todo 4
        </span>
      </span>
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
    </div>
  </li>
  <li
    className="list-group-item false"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="d-flex align-items-center"
      >
        <input
          className="mr-3"
          defaultChecked={false}
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="ml-1 mr-1"
        >
          todo 5
        </span>
      </span>
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
    </div>
  </li>
</ul>
`;
